UnitySDK/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs

```diff
@@ -82,36 +84,37 @@ public override void AgentAction(float[] vectorAction, string textAction)
         Vector3 targetPos = transform.position;
         switch (action)
         {
-            case NoAction:
+            case k_NoAction:
                 // do nothing
                 break;
-            case Right:
+            case k_Right:
                 targetPos = transform.position + new Vector3(1f, 0, 0f);
                 break;
-            case Left:
+            case k_Left:
                 targetPos = transform.position + new Vector3(-1f, 0, 0f);
                 break;
-            case Up:
+            case k_Up:
                 targetPos = transform.position + new Vector3(0f, 0, 1f);
                 break;
-            case Down:
+            case k_Down:
                 targetPos = transform.position + new Vector3(0f, 0, -1f);
                 break;
             default:
                 throw new ArgumentException("Invalid action value");
         }
 
-        Collider[] blockTest = Physics.OverlapBox(targetPos, new Vector3(0.3f, 0.3f, 0.3f));
-        if (blockTest.Where(col => col.gameObject.CompareTag("wall")).ToArray().Length == 0)
+
+        Physics.OverlapBoxNonAlloc(targetPos, new Vector3(0.3f, 0.3f, 0.3f), m_BlockTest);
```

Comment:
Guessing this wasn't Rider? :)  I'd say premature optimization, but if you think it's better, OK to leave.

Reply:
It actually was a Rider suggestion.  I had to create the `m_BlockTest` array myself though